### PR TITLE
Add documentation to models and views in backend

### PR DESCRIPTION
**Pull request by owenl131**

### CODE REVIEW

File: api/views.py [hunk 1]
```diff
@@ -436,6 +537,14 @@ def root_settings(request):
 
 @api_view(['POST'])
 def upload_sample_input(request):
+    """
+    Creates a sample input. Only image input is supported currently.
+
+    The data passed into the POST request needs to contain:
+        demo_id (str): The demo that the sample input should belong to.
+        sample-image* (file): A value with its key having 'sample-image'
```
Comment (AvaisP) [inline api/views.py]: sample-image-*

**Comment by AvaisP** [comment]
Great job on the views! 
Please describe all the fields of the model

File: api/views.py (hunk 1 above)
Comment (owenl131) [reply api/views.py]: Hmm but sample-image is the prefix required, not sample-image-

**Comment by owenl131** [comment]
Ok, will do

File: api/views.py (hunk 1 above)
Comment (owenl131) [reply api/views.py]: Actually I'm not at all sure how it should be written, so just wrote some regex like thing

Comment (owenl131) [reply api/views.py]: But that makes more sense actually

File: api/views.py [hunk 2]
```diff
@@ -152,6 +175,17 @@ def get_all_demos(request):
 
 @api_view(['GET', 'POST', 'PUT', 'DELETE'])
 def custom_component_controller(request, type_req, user_id, demoid):
+    """
+    Gets the properties of, adds, updates, or removes an
+    InputComponent or OutputComponent given a
+    GET, POST, PUT or DELETE request respectively.
+
+    Args:
```
Comment (AvaisP) [inline api/views.py]: Keyword arguments

**Comment by owenl131** [comment]
@AvaisP I think travis had some internet issues. The tests should be fine.

**Pull request merged**